apis/actions.summerwind.net/v1alpha1/runner_types.go

```diff
@@ -248,10 +248,60 @@ type RunnerStatus struct {
 	// +optional
 	Message string `json:"message,omitempty"`
 	// +optional
+	WorkflowStatus *WorkflowStatus `json:"workflow"`
```

Comment:
In your original work, you directly modified the auto-generated CRD. This is how you should add fields in Go, which are then reflected to the (re-)generated CRDs by running `make manifests`!

I've also grouped the all the workflow related status fields into the `status.workflow` field so that it's easy to extract only workflow related statuses for further processing.
For example, you can use `kubectl get runner --watch -o json | jq .status.worfklow` so obtain the workflow status stream only.

Reply:
Thanks for the explanation, this way looks way better